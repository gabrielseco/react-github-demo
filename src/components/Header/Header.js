// @flow
import React from 'react';
import styles from './Header.scss';

const Header = () => (
  <header className={styles.header}>
    <h2>Github Demo</h2>
  </header>
);

export default Header;
