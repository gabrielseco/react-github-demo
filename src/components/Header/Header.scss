$header-min-height: 4.6875em;
$header-background-color: #ffffff;
$header-color: #212240;
$header-border-color: rgba(33, 34, 64, 0.05);
$header-box-shadow-color: rgba(50, 50, 50, 0.05);

$header-logo-padding: 0 20px;

.header {
  align-items: center;
  background: $header-background-color;
  border-bottom: 1px solid $header-border-color;
  box-shadow: 0 5px 15px 1px $header-box-shadow-color;
  color: $header-color;
  display: flex;
  justify-content: space-between;
  min-height: $header-min-height;
}

.logo {
  flex: 3;
  padding: $header-logo-padding;
}

.title {}
