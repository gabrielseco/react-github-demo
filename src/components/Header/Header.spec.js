import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Header from './Header';

describe('Header suite', () => {
  it('should render the Header component', () => {
    const component = shallow(<Header />);
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>
      )
      .toJSON();

    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
