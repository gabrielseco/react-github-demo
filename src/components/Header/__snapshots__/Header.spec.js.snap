// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header suite should render the Header component 1`] = `
<header
  className="header"
>
  <h2>
    Github Demo
  </h2>
</header>
`;
